commons/src/lib.rs

```diff
@@ -91,16 +91,34 @@ pub fn ensure_query_params(
     Ok(())
 }
 
-/// Make sure client requested a valid content type.
-pub fn ensure_content_type(
+/// Make sure the client can accept the provided media type.
+pub fn validate_content_type(
     headers: &HeaderMap,
     content_type: &'static str,
 ) -> Result<(), GraphError> {
-    let content_json = header::HeaderValue::from_static(content_type);
+    if !headers.contains_key(header::ACCEPT) {
+        return Ok(());
+    }
+
+    let full_type = header::HeaderValue::from_static(content_type);
+    let wildcard = header::HeaderValue::from_static("*");
+    let double_wildcard = header::HeaderValue::from_static("*/*");
+    let top_type = content_type.split("/").next().unwrap_or("");
+    let top_type_wildcard = header::HeaderValue::from_str(&format!("{}/*", top_type));
+    assert!(
```

Comment:
we should be throwing `Err(GraphError::InvalidContentType)` here:
```
if !top_type_wildcard.is_ok() { 
    ...
```

Reply:
I wanted to panic here instead of returning a `GraphError`, because it's not the client's fault if we can't construct `top_type_wildcard`.  Is there a better way to distinguish between "dev(-ops) issue" and "client issue" besides `assert` panics?

Reply:
>it's not the client's fault if we can't construct top_type_wildcard

The only input for it is `content_type`, so its certainly client's fault if error occurs here

Reply:
> The only input for it is `content_type`, so its certainly client's fault if error occurs here

Client is responsible for `headers`.  We are responsible for `content_type`:

```console
$ git --no-pager grep commons::ensure_content_type origin/master
origin/master:graph-builder/src/graph.rs:    commons::ensure_content_type(req.headers(), CONTENT_TYPE)?;
origin/master:policy-engine/src/graph.rs:    commons::ensure_content_type(req.headers(), CONTENT_TYPE)?;
$ git --no-pager grep 'const CONTENT_TYPE' origin/master
origin/master:cincinnati/src/lib.rs:pub const CONTENT_TYPE: &str = "application/json";
```